mas-foundation/src/main/java/com/ca/mas/core/service/MssoService.java

```diff
@@ -41,19 +47,50 @@
  * An JobIntentService that receives outbound HTTP requests encoded into Intents and returns the eventual responses
  * via a ResultReceiver.
  */
-public class MssoService extends JobIntentService {
+public class MssoService extends Service {
 
-    private static final int JOB_ID = 1000;
+    private final IBinder binder = new MASBinder();
+    AtomicInteger threadNum = new AtomicInteger(1);
```

Comment:
Why is Atomic Integer used here ? Since atomic integer's atomic is not being used (it is just used as initialization of thread), hence you can replace it with normal int thereby avoiding the overhead

Reply:
Usage was supposed to be threadNum.getAndIncrement(). But as you pointed out, this varaible itself can be removed, as only one thread will be created. Hence removing the variable.